From a7f0e350acbaad44040aefde7b637722ddeef150 Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert@linux-m68k.org>
Date: Sun, 29 Mar 2009 21:15:40 +0200
Subject: [PATCH] m68k: z2ram - correct printing of sector_t

If CONFIG_LBDAF=y, `sector_t' becomes `u64' instead of `unsigned long':

| drivers/block/z2ram.c:79: warning: format '%lu' expects type 'long unsigned int', but argument 2 has type 'sector_t'

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
---
 drivers/block/z2ram.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/block/z2ram.c b/drivers/block/z2ram.c
index 64f941e..1d336f9 100644
--- a/drivers/block/z2ram.c
+++ b/drivers/block/z2ram.c
@@ -77,8 +77,9 @@ static void do_z2_request(struct request_queue *q)
 		int err = 0;
 
 		if (start + len > z2ram_size) {
-			printk( KERN_ERR DEVICE_NAME ": bad access: block=%lu, count=%u\n",
-				blk_rq_pos(req), blk_rq_cur_sectors(req));
+			pr_err(DEVICE_NAME ": bad access: block=%llu, count=%u\n",
+				(unsigned long long)blk_rq_pos(req),
+				blk_rq_cur_sectors(req));
 			err = -EIO;
 			goto done;
 		}
-- 
1.5.6.5

